### DISCUSSION TITLE
testcontainers 3.0.0 MySqlBuilder Docker.DotNet.DockerApiException

### DISCUSSION THREAD

**Discussion by MikeFXR** [testcontainers/testcontainers-dotnet#844] [Q&A] [answered]
Hello,

I'm trying to get testcontainers 3.0.0 using the MySqlContainer module working.

`private readonly MySqlContainer _dbContainer = new MySqlBuilder().Build();
`
`await _dbContainer.StartAsync().ConfigureAwait(false);
`

I get the error
```
Docker.DotNet.DockerApiException : Docker API responded with status code=InternalServerError, response={"message":"Container 6d998daf19c2556b64582346fca7162d8538e1696e63cc22bc60edf424110b04 is not running: Exited (1) Less than a second ago"}
```

I see the containers spin up in docker but then it fails!

The reason I'm trying to test out 3.0.0 is because I'm getting the same issue with 2.1.0, which was working fine in our project up until a  couple of days ago. Nothing appears to have changed, it's just stopped working!

FYI this is the 2.1.0 code we're using 
```
private readonly MySqlTestcontainer _dbContainer = new TestcontainersBuilder<MySqlTestcontainer>()
  .WithDatabase(new MySqlTestcontainerConfiguration
  {
      Database = "Test",
      Username = "Test",
      Password = "Test",
      Port = new Random().Next(7000, 8000)
  }).WithCleanUp(true).Build();
```

If anyone could shed any light on the error that would be great.
Thanks

**Comment by HofmeisterAn** [comment, marked as answer]
> The reason I'm trying to test out 3.0.0 is because I'm getting the same issue with 2.1.0, which was working fine in our project up until a couple of days ago.

This sounds like something in your environment has changed. What happens if you run a container manual:

```console
docker run -e MYSQL_ROOT_PASSWORD=my-secret-pw -d mysql:8.0
```

FYI: I would not recommend assigning a random port like you do in your example. This is not safe at all.

**Reply by MikeFXR** [reply]
`docker run -e MYSQL_ROOT_PASSWORD=my-secret-pw -d mysql:8.0` so this failed and I found and stopped the process running on TCP:3306. Running the container manually now works, thanks!

I now get this error in the test `DotNet.Testcontainers.Containers.ResourceReaperException : Initialization has been cancelled.` on `StartAsync` 

```
Stack Trace: 
  ResourceReaper.GetAndStartNewAsync(Guid sessionId, IDockerEndpointAuthenticationConfiguration dockerEndpointAuthConfig, String ryukImage, TimeSpan initTimeout, CancellationToken ct)
  ResourceReaper.GetAndStartNewAsync(Guid sessionId, IDockerEndpointAuthenticationConfiguration dockerEndpointAuthConfig, String ryukImage, TimeSpan initTimeout, CancellationToken ct)
  ResourceReaper.GetAndStartDefaultAsync(IDockerEndpointAuthenticationConfiguration dockerEndpointAuthConfig, CancellationToken ct)
  TestcontainersClient.RunAsync(ITestcontainersConfiguration configuration, CancellationToken ct)
  TestcontainersContainer.Create(CancellationToken ct)
  TestcontainersContainer.StartAsync(CancellationToken ct)
  GlobalTrainingPlansApplication.InitializeAsync() line 80
```

>FYI: I would not recommend assigning a random port like you do in your example. This is not safe at all.

Why is this bad? What would you suggest doing?

Thanks again

**Reply by HofmeisterAn** [reply]
> I now get this error in the test `DotNet.Testcontainers.Containers.ResourceReaperException : Initialization has been cancelled.` on `StartAsync`

Can you please use the latest version 3.0.0 ([Testcontainers.MySql](https://www.nuget.org/packages/Testcontainers.MySql) module)?

> Why is this bad? What would you suggest doing?

That does not protect from ruining into ports that are already in use (port clashes), see our [best practices](https://dotnet.testcontainers.org/api/best_practices/). Testcontainers handles the random port assignment automatically. There is no need to configure it.

**Reply by MikeFXR** [reply]
> Can you please use the latest version 3.0.0 ([Testcontainers.MySql](https://www.nuget.org/packages/Testcontainers.MySql) module)?

This seems to have fixed the issue. Not sure why it stopped working in the first place though. 
Thanks for quick responses
